### ISSUE TITLE
Make sure `wheel` is installed at the beginning of `update_package.sh`

### ISSUE THREAD

**Issue by christophertubbs** [NOAA-OWP/DMOD#680]
I just launched a clean dev environment and `update_package.sh` broke because the setup calls `bdist_wheel`. Fixing it is easy - `pip install wheel` does the trick.

`update_package.sh` should check for `wheel` and install it if it's not there _or_ it should fail if it's not there. _**I**_ figured out that it was `wheel`, but the error would be confusing for the uninitiated.

**Comment by aaraney** [comment]
Yeah, I think moving to `pyproject.toml` would resolve this, but also cause us to make a few changes. The issue is that with `setup.py` we have no way of specifying build time deps, so we have to have, in this case, `wheel` installed first. Super annoying for sure.

**Comment by robertbartel** [comment]
I think we still need to answer the question of which correction is most appropriate:  install or more informative error.

My first thought was "install," and to just approve #683, but there is already an option for the script to install project dependencies (`--dependencies` or `-d`) from the project requirements file.  I lean now more toward an error message that also suggests that optional param.  If nothing else, we avoid repeating where the dependency on wheel is defined.

Are there any compelling reasons, beyond slightly more convenience, why auto-installing is advantageous?

**Comment by aaraney** [comment]
> My first thought was "install," and to just approve #683, but there is already an option for the script to install project dependencies (`--dependencies` or `-d`) from the project requirements file.

I think we should go with something like #683 (more on that later). `wheel` isn't a project dependency, it's a dependency of the build script, `update_package.sh`. You can technically install `dmod` packages without explicitly installing `wheel` its just not very convenient. 

Aside, I think we should use the python [`build`](https://packaging.python.org/en/latest/key_projects/#build) module instead of [`python setup.py sdist bdist_wheel`](https://github.com/NOAA-OWP/DMOD/blob/6f39f1139bb5561ade060b061a9f50a24632dbe2/scripts/update_package.sh#L256) anyways. [Calling `setup.py` like a script is deprecated](https://packaging.python.org/en/latest/discussions/setup-py-deprecated/#should-setup-py-be-deleted). We should move to `pyproject.toml` over `setup.py` anyways and moving to `build` will ease the transition.

**Comment by christophertubbs** [comment]
If we're going to have a helper script that installs everything, it shouldn't fail because it hasn't hit code that hasn't installed the stuff that enables installation or that it has been installed from the get-go. If we're going to have a "hey, let me install that for you" script, it should either ensure that it is capable of installing its own dependencies out of the box or allow the process to run with vanilla capabilities - i.e. if it requires a `pip install`, it's off limits.

As far as requiring `--dependencies`, I wouldn't for a second think that I would have to tell the script to install the dependencies for all of the libraries in order to make sure that the library for the script itself is available.

As far as the question of "should we install the requirement or throw an error", it literally takes just as much if not less to install the required and pretty uncontroversial library. If it were something like pydantic, ok, user needs to make a decision, but it's `wheel`. Halting operation to tell the user to rerun the application with an additional flag when it _already_ knows how to resolve the situation is over the top.

I agree with Austin in that a lot of these issues will probably go away if we stopped using functionality that was deprecated in 2021.

**Comment by robertbartel** [comment]
You both have me convinced.  A couple follow up questions:

> `wheel` isn't a project dependency, it's a dependency of the build script, update_package.sh.

Should we (as part of this or a separate issue) remove `wheel` from _requirements.txt_?

> As far as requiring `--dependencies`, I wouldn't for a second think that I would have to tell the script to install the dependencies for all of the libraries in order to make sure that the library for the script itself is available.

Should we (again, as part of this or a separate issue) take out the `--dependencies` option and just have the script make sure this is done?

**Comment by christophertubbs** [comment]
> Should we (again, as part of this or a separate issue) take out the --dependencies option and just have the script make sure this is done?

I would say that it should pretty much always evaluate whether dependencies need to be installed. Nothing really happens if an install like that is up to date and what's the point of trying to install libraries if the lack of their dependencies prevent them from functioning?

**Comment by PhilMiller** [comment]
The consideration would seem to be whether you want/need a script that installs just the DMOD components themselves, in what should otherwise be a functional Python environment, that the user doesn't want you to mess with. Say, because they have the dependencies for other stuff they want to use together with DMOD all worked out.
